src/core/include/openvino/core/graph_util.hpp

```diff
@@ -234,18 +244,25 @@ std::vector<std::shared_ptr<Node>> topological_sort(T root_nodes) {
         if (nodes_done.count(node) == 0) {
             bool can_add = true;
             size_t arg_count = node->get_input_size();
+            size_t node_input_degree = arg_count + node->get_control_dependencies().size();
```

Comment:
I believe that node_degree is a count of outputs, not inputs

Reply:
I changed the logic - now there is no need to track edges at all